gcc/rust/expand/rust-macro-builtins.cc

```diff
@@ -109,13 +108,12 @@ std::unordered_map<std::string, AST::MacroTranscriberFunc>
     {"include", MacroBuiltin::include_handler},
     {"format_args", format_args_maker (AST::FormatArgs::Newline::No)},
     {"format_args_nl", format_args_maker (AST::FormatArgs::Newline::Yes)},
+    {"asm", MacroBuiltin::nonglobal_asm_handler},
+    {"global_asm", MacroBuiltin::global_asm_handler},
```

Comment:
here you can use the same technique as we did for `format_args_maker` - because both `nonglobal_asm_handler` and `global_asm_handler` call into `parse_asm`, simply with a different last parameter, it's cleaner to have just one function/lambda and it's easier to maintain

Reply:
resolved